nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GetChangeDataCaptureMySQL.java

```diff
@@ -0,0 +1,879 @@
+     * @return List of InetSocketAddresses for the hosts
+     */
+    private List<InetSocketAddress> getHosts(String hostsString) {
+
+        if (hostsString == null) {
+            return null;
+        }
+        final List<String> hostsSplit = Arrays.asList(hostsString.split(","));
+        List<InetSocketAddress> hostsList = new ArrayList<>();
+
+        for (String item : hostsSplit) {
+            String[] addresses = item.split(":");
+            if (addresses.length != 2) {
+                throw new ArrayIndexOutOfBoundsException("Not in host:port format");
+            }
+
+            hostsList.add(new InetSocketAddress(addresses[0].trim(), Integer.parseInt(addresses[1].trim())));
+        }
+        return hostsList;
+    }
+
+    protected void connect(List<InetSocketAddress> hosts, String username, String password, boolean createEnrichmentConnection,
+                           String driverLocation, String driverName, long connectTimeout) throws IOException {
+
+        int connectionAttempts = 0;
+        final int numHosts = hosts.size();
+        InetSocketAddress connectedHost = null;
+        ComponentLog log = getLogger();
+
+        while (connectedHost == null && connectionAttempts < numHosts) {
+            if (binlogClient == null) {
+
+                connectedHost = hosts.get(currentHost);
+                binlogClient = createBinlogClient(connectedHost.getHostString(), connectedHost.getPort(), username, password);
+            }
+
+            BinlogEventListener eventListener = createBinlogEventListener(binlogClient, queue);
+            binlogClient.registerEventListener(eventListener);
+
+            binlogClient.setBinlogFilename(currentBinlogFile);
+            binlogClient.setBinlogPosition(currentBinlogPosition);
+
+            try {
+                if (connectTimeout == 0) {
+                    connectTimeout = Long.MAX_VALUE;
+                }
+                binlogClient.connect(connectTimeout);
+
+            } catch (IOException | TimeoutException te) {
+                // Try the next host
+                connectedHost = null;
+                currentHost = (currentHost + 1) % numHosts;
+                connectionAttempts++;
+            }
+        }
+        if (!binlogClient.isConnected()) {
+            throw new IOException("Could not connect to any of the specified hosts");
+        }
+
+        if (createEnrichmentConnection) {
+            try {
+                jdbcConnection = getJdbcConnection(driverLocation, driverName, connectedHost, username, password, null);
+            } catch (InitializationException | SQLException e) {
+                throw new ProcessException(e);
+            }
+        }
+
+        doStop.set(false);
+    }
+
+
+    public void outputEvents(ProcessSession session, ComponentLog log) throws IOException {
+        RawBinlogEvent rawBinlogEvent;
+
+        // Drain the queue
+        ;
+        while ((rawBinlogEvent = queue.poll()) != null && !doStop.get()) {
```

Comment:
It does not, nor does it unregister the event listeners, so we might have a memory leak there. I will look into this and call stop() on our listener as well as unregisterEventListener() if need be. I may have to introduce a LifecycleListener to ensure the correct order of operations on cleanup.  Nice catch, thanks!